submission_worker.py

```diff
@@ -0,0 +1,300 @@
+django.setup()
+
+from challenges.models import Challenge     # noqa
+from jobs.models import Submission          # noqa
+
+
+BASE_CHALLENGE_DATA_DIR = join(os.getcwd(), 'challenge_data')
+BASE_SUBMISSION_DATA_DIR = join(os.getcwd(), 'submission_files')
+CHALLENGE_DATA_DIR = join(BASE_CHALLENGE_DATA_DIR, '{challenge_id}')
+PHASE_DATA_DIR = join(os.getcwd(), 'challenge_data', '{challenge_id}', 'phase_data', '{phase_id}')
+SUBMISSION_DATA_DIR = join(BASE_SUBMISSION_DATA_DIR, '{submission_id}')
+
+django.db.close_old_connections()
+
+
+class ExecutionTimeLimitExceeded(Exception):
+    pass
+
+
+def alarm_handler(signum, frame):
+    raise ExecutionTimeLimitExceeded
+
+
+def download_and_extract_file(url, download_location, extract_location):
+    '''
+        * Function to extract downlaod a zip file, extract it and then removes the zip file.
+        * `download_location` should include name of file as well.
+    '''
+    try:
+        response = requests.get(url)
+    except Exception as e:
+        print 'Failed to fetch file from {}, error {}'.format(url, e)
+        # return from here or not?
+        response = None
+
+    if response and response.status_code == 200:
+        with open(download_location, 'w') as f:
+            f.write(response.content)
+        # extract zip file
+        zip_ref = zipfile.ZipFile(download_location, 'r')
+        zip_ref.extractall(extract_location)
+        zip_ref.close()
+        # delete zip file
+        try:
+            os.remove(download_location)
+        except Exception as e:
+            print 'Failed to remove zip file {}, error {}'.format(download_location, e)
+
+
+def create_dir(directory):
+    '''
+        Creates a directory if it does not exists
+    '''
+    if not os.path.exists(directory):
+        os.makedirs(directory)
+
+
+def extract_challenge_data(challenge, phases):
+    '''
+        * Expects a challenge object and an array of phase object
+        * Extracts `evaluation_script` for challenge and `annotation_file` for each phase
+
+    '''
+
+    challenge_data_directory = CHALLENGE_DATA_DIR.format(challenge_id=challenge.id)
+    evaluation_script_url = challenge.evaluation_script
+    # create challenge directory
+    create_dir(challenge_data_directory)
+
+    challenge_zip_file = join(challenge_data_directory, 'challenge_{}.zip'.format(challenge.id))
+    download_and_extract_file(evaluation_script_url, challenge_zip_file, challenge_data_directory)
+
+    phase_data_base_directory = join(challenge_data_directory, 'phase_data')
+
+    for phase in phases:
+        phase_data_directory = join(phase_data_base_directory, str(phase.id))
+        # create phase directory
+        create_dir(phase_data_directory)
+        phase_zip_file = join(phase_data_directory, 'phase_{}.zip'.format(phase.id))
+        annotation_file_url = phase.test_annotation
+        download_and_extract_file(annotation_file_url, phase_zip_file, phase_data_directory)
```

Comment:
I think the annotation file will just be a single file containing the ground truth. In the current setup, it won't be a zip file. 
@deshraj : please confirm?